include/drivers/uart.h

```diff
@@ -467,511 +488,223 @@ __subsystem struct uart_driver_api {
 
 };
 
+/** @endcond */
 
 /**
- * @brief Set event handler function.
- *
- * Since it is mandatory to set callback to use other asynchronous functions,
- * it can be used to detect if the device supports asynchronous API. Remaining
- * API does not have that detection.
+ * @brief Check whether an error was detected.
  *
- * @param dev       UART device structure.
- * @param callback  Event handler.
- * @param user_data Data to pass to event handler function.
+ * @param dev UART device structure.
  *
- * @retval -ENOSYS  If not supported by the device.
- * @retval -ENOTSUP If API not enabled.
- * @retval 0	    If successful, negative errno code otherwise.
+ * @retval uart_rx_stop_reason If error during receiving occurred.
```

Comment:
format is `@retval <value> <description>`, value and description here don't provide much information. Does this really mean:

```c
@retval 0 If success
@retval -ENOSYS If not implemented
@retval -errno Other negative errno value in case of failure
```
?

Reply:
It returns error flags set as defined in `enum uart_rx_stop_reason`, so it's not an `errno`. Suggestion:
```
 * @retval 0 If no error was detected
 * @retval err Error flags as defined in @ref uart_rx_stop_reason
 * @retval -ENOSYS If not implemented
```

Reply:
sounds good, will resolve to a reference